src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,471 @@
+            UpdatePseudoClasses();
+        }
+
+        /// <inheritdoc/>
+        protected override void OnAttachedToLogicalTree(LogicalTreeAttachmentEventArgs e)
+        {
+            base.OnAttachedToLogicalTree(e);
+
+            if (Command != null)
+            {
+                Command.CanExecuteChanged += CanExecuteChanged;
+                CanExecuteChanged(this, EventArgs.Empty);
+            }
+
+            _isAttachedToLogicalTree = true;
+        }
+
+        /// <inheritdoc/>
+        protected override void OnDetachedFromLogicalTree(LogicalTreeAttachmentEventArgs e)
+        {
+            base.OnDetachedFromLogicalTree(e);
+
+            if (Command != null)
+            {
+                Command.CanExecuteChanged -= CanExecuteChanged;
+            }
+
+            _isAttachedToLogicalTree = false;
+        }
+
+        /// <inheritdoc/>
+        protected override void OnPropertyChanged<T>(AvaloniaPropertyChangedEventArgs<T> e)
+        {
+            if (e.Property == CommandProperty)
+            {
+                if (_isAttachedToLogicalTree)
+                {
+                    // Must unregister events here while a reference to the old command still exists
+                    if (e.OldValue.GetValueOrDefault() is ICommand oldCommand)
+                    {
+                        oldCommand.CanExecuteChanged -= CanExecuteChanged;
+                    }
+
+                    if (e.NewValue.GetValueOrDefault() is ICommand newCommand)
+                    {
+                        newCommand.CanExecuteChanged += CanExecuteChanged;
+                    }
+                }
+
+                CanExecuteChanged(this, EventArgs.Empty);
+            }
+            else if (e.Property == CommandParameterProperty)
+            {
+                CanExecuteChanged(this, EventArgs.Empty);
+            }
+            else if (e.Property == FlyoutProperty)
+            {
+                var oldFlyout = e.OldValue.GetValueOrDefault() as FlyoutBase;
+                var newFlyout = e.NewValue.GetValueOrDefault() as FlyoutBase;
+
+                // If flyout is changed while one is already open, make sure we 
+                // close the old one first
+                // This is the same behavior as Button
+                if (oldFlyout != null &&
+                    oldFlyout.IsOpen)
+                {
+                    oldFlyout.Hide();
+                }
+
+                // Must unregister events here while a reference to the old flyout still exists
+                if (oldFlyout != null)
+                {
+                    UnregisterFlyoutEvents(oldFlyout);
+                }
+
+                if (newFlyout != null)
+                {
+                    RegisterFlyoutEvents(newFlyout);
+                }
+
+                UpdatePseudoClasses();
+            }
+
+            base.OnPropertyChanged(e);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnKeyDown(KeyEventArgs e)
```

Comment:
@maxkatz6 You removed the _onKeyDown variable and also removed that functionality from the style. This is important functionality that should exist. When the user presses the Enter key they should have visual feedback the same as when a button is pressed... I'm really not happy to have to go through and retest things like this. We've somehow switched places in the PR.

I do not know how to add that functionality back. It would require setting the :pressed PseudoClass of the Primary button. 

Additionally I've noted the Button uses space instead of enter. I will probably add support for both.

Reply:
> I do not know how to add that functionality back

We have PressedMixin class which does it for other controls, where we need ":pressed" class for pointer input.
Can be extended to work with keyboard input. 
Alternatively, this pseudoclass can be set in SplitButton. Or it can just work if it will inherit Button/ButtonBase.

> We've somehow switched places in the PR.

It will be fair, if I will fix this part. Also, I can fix issue from [this comment](https://github.com/AvaloniaUI/Avalonia/pull/7422#discussion_r793139596) and add unused Border for styling flexibility from [this comment](https://github.com/AvaloniaUI/Avalonia/pull/7422#discussion_r793136969). Just tell me if you agree.

Reply:
Thanks, let me do a round of changes we discussed first. I think adding back :pressed to SplitButton and then adding a Background Border again in the style as discussed separately will take care of this. Still not a fan of inheriting from Button quite yet.

My changes should done by tomorrow afternoon at the latest. Then I can pass it back to you to take a final look here.

Reply:
@maxkatz6  I'll leave the updates here to you after my comments elsewhere. The assumption is still we need to add back the ability to manage presses for the entire control at once: :pressed pseudoclass added back, key press handling code-behind added back, along with adding back the background RootBorder.

Reply:
> add back the ability to manage presses

:pressed pseudoclass for a whole control wasn't there before in this PR.
I will add it.